packages/fluentui/react-context-selector/src/useContextSelector.ts

```diff
@@ -18,39 +12,70 @@ export const useContextSelector = <Value, SelectedValue>(
   context: Context<Value>,
   selector: ContextSelector<Value, SelectedValue>,
 ): SelectedValue => {
-  const { subscribe, value } = React.useContext((context as unknown) as Context<ContextValue<Value>>);
-  const [, forceUpdate] = React.useReducer((c: number) => c + 1, 0) as [never, () => void];
+  const contextValue = React.useContext((context as unknown) as Context<ContextValue<Value>>);
 
-  const ref = React.useRef<UseSelectorRef<Value, SelectedValue>>();
+  const {
+    STRUCT: {
+      value: { current: value },
+      version: { current: version },
+      listeners,
+    },
+  } = contextValue;
   const selected = selector(value);
 
-  useIsomorphicLayoutEffect(() => {
-    ref.current = {
-      selector,
-      value,
-      selected,
-    };
-  });
-  useIsomorphicLayoutEffect(() => {
-    const callback = (nextState: Value) => {
+  const [state, dispatch] = React.useReducer(
```

Comment:
would this be more appropriate ? 
```suggestion
  const [state, rerender] = React.useReducer(
```

Reply:
It's not `rerender` as it _will not perform_ re-render

Reply:
I think `dispatch` is fine, it's what `useReducer` return

Reply:
oh really ? it is literally commented with `schedule a rerender`